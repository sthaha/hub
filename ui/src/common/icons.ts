// eslint-disable-next-line
enum Icons {
  Unknown = 'none',
  Cat = 'cat',
  Certificate = 'certificate',
  User = 'user',
  Build = 'build',
  Domain = 'domain'
}

export default Icons;
