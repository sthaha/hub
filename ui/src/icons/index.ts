export enum IconName {
  none = 'none',
  cat = 'cat',
  certificate = 'certificate',
  user = 'user',
  build = 'build',
  domain = 'domain'
}
