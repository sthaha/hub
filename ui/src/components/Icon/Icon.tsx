import React from 'react';
import {
  CatIcon,
  CertificateIcon,
  BuildIcon,
  DomainIcon,
  UserIcon,
  IconSize
} from '@patternfly/react-icons';

import { IconName } from '../../icons';
import './Icon.css';

export interface IconProps {
  id: IconName | keyof IconName;
  size: IconSize | keyof typeof IconSize;
  label: string;
}

const Icon: React.FC<IconProps> = (props: IconProps) => {
  const { id, size, label } = props;
  switch (id) {
    case IconName.cat:
      return <CatIcon size={size} className="hub-icon" label={label} />;
    case IconName.certificate:
      return <CertificateIcon size={size} className="hub-icon" label={label} />;
    case IconName.user:
      return <UserIcon size={size} className="hub-icon" label={label} />;
    case IconName.build:
      return <BuildIcon size={size} className="hub-icon" label={label} />;
    case IconName.domain:
      return <DomainIcon size={size} className="hub-icon" label={label} />;
  }
  return <div></div>;
};

export default Icon;
