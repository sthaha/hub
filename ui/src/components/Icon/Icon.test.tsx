import React from 'react';
import { shallow } from 'enzyme';
import Icon from './Icon';
import { IconName } from '../../icons';

describe('Icon Component', () => {
  it('should render icon for Task Kind Filter', () => {
    const component = shallow(<Icon id={IconName.cat} size="sm" label="Task" />);
    expect(component).toMatchSnapshot();
    expect(component.find('CatIcon[label="Task"]').length).toEqual(1);
  });
  it('should render icon for Pipeline Kind Filter', () => {
    const component = shallow(<Icon id={IconName.build} size="md" label="Pipeline" />);
    expect(component).toMatchSnapshot();
    expect(component.find('BuildIcon[label="Pipeline"]').length).toEqual(1);
  });
  it('should render icon for Official Catalog Filter', () => {
    const component = shallow(<Icon id={IconName.domain} size="lg" label="Official" />);
    expect(component).toMatchSnapshot();
    expect(component.find('DomainIcon[label="Official"]').length).toEqual(1);
  });
  it('should render icon for Verified Catalog Filter', () => {
    const component = shallow(<Icon id={IconName.certificate} size="xl" label="Verified" />);
    expect(component).toMatchSnapshot();
    expect(component.find('CertificateIcon[label="Verified"]').length).toEqual(1);
  });
  it('should render icon for Community Catalog Filter', () => {
    const component = shallow(<Icon id={IconName.user} size="sm" label="Community" />);
    expect(component).toMatchSnapshot();
    expect(component.find('UserIcon[label="Community"]').length).toEqual(1);
  });
  it('should render icon for Category Filter', () => {
    const component = shallow(<Icon id={IconName.none} size="sm" label="CLI" />);
    expect(component).toMatchSnapshot();
  });
});
