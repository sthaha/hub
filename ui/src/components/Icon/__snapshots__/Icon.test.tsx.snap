// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component should render icon for Category Filter 1`] = `ShallowWrapper {}`;

exports[`Icon Component should render icon for Community Catalog Filter 1`] = `ShallowWrapper {}`;

exports[`Icon Component should render icon for Official Catalog Filter 1`] = `ShallowWrapper {}`;

exports[`Icon Component should render icon for Pipeline Kind Filter 1`] = `ShallowWrapper {}`;

exports[`Icon Component should render icon for Task Kind Filter 1`] = `ShallowWrapper {}`;

exports[`Icon Component should render icon for Verified Catalog Filter 1`] = `ShallowWrapper {}`;
